lib/bind.js

```diff
@@ -19,7 +19,16 @@ function bind(fn, context, args) {
     assert.optionalArray(args, 'args');
 
     return function boundFn() {
-        fn.apply(context || null, args || []);
+
+        // working with arguments is a bit hazardous in terms of deopt.
+        // https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments
```

Comment:
What about the third method outlined here:
http://jsperf.com/arguments-leaking/3
It looks like it might be faster in v8 (well, chrome anyway) to use `Array.apply(null, arguments)` instead.